app/models/concerns/edm/entity/referenced_records.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+module EDM
+  module Entity
+    module ReferencedRecords
+      include RepositoryHelper
+      extend ActiveSupport::Concern
+
+      def unreferenced?
+        !referenced_records.detect { |_key, records| records[:total][:value] != 0 }
+      end
+
+      def referenced_records
+        @refereneced_records ||= {}
+        search_keys.each do |key|
+          @refereneced_records[key] ||= begin
+            return { search_results: [], total: { value: 0, formatted: '0' } } unless respond_to?(:search_query)
+            @response = Europeana::Blacklight::Response.new(repository.search(query: search_query), controller.params)
+            {
```

Comment:
This is used for display purposes, so move it into the `Entities::Show` view class.